### PR TITLE
improved submitted DocId error

### LINKED ISSUE #509: SubmittedIdDoesNotMatchBase error is cryptic
As of f58d7a00bb321486e79cb8913556df5f00cfd42a, we have a wonderful new error message. I'm guessing it has to do with a mismatch between the `@key` `@type` and the provided `@id`, but it is difficult to understand. Can we improve on the error message?

Script (run from a freshly initialized store):

```sh
#!/bin/bash

set -ex

# Create the database.
xh 'http://admin:root@localhost:6363/api/db/admin/t' <<EOF
{"label":"l","comment":"c"}
EOF

# Create the schema.
xh 'http://admin:root@localhost:6363/api/document/admin/t?author=a&message=m&graph_type=schema' <<EOF
{"@type": "Class","@id": "String"}
EOF

# Post an instance.
xh 'http://admin:root@localhost:6363/api/document/admin/t?author=a&message=m' <<EOF
{"@id": "something","@type": "String"}
EOF
```

Output of last command:

```json
HTTP/1.1 400 Bad Request
connection: keep-alive
content-length: 402
content-type: application/json
date: Thu, 02 Sep 2021 17:47:32 GMT

{
    "@type": "api:InsertDocumentErrorResponse",
    "api:error": {
        "@type": "api:SubmittedIdDoesNotMatchBase",
        "api:base": "String/",
        "api:document": {
            "@id": "something",
            "@type": "String"
        },
        "api:submitted_id": "terminusdb:///data/something"
    },
    "api:message": "Document was submitted with id 'terminusdb:///data/something' which does not match base 'String/'",
    "api:status": "api:failure"
}
```

Some questions that arise from reading the above:
* What is `api:base`? Does it have anything to do with `@base`? If so, how? If not, then perhaps it should be called something else.
* Why does `String/` have the `/` at the end?
* Where did `terminusdb:///data/something` come from? I gave the `@id` `something`, but why am I seeing `terminusdb:///data/` in front of it? If `terminusdb:///data/` is the default `@base`, should `String/` also have `terminusdb:///data/` in front of it?
* Is `terminusdb:///data/something` is supposed to match `String/`? How?

### LINKED ISSUE #528: SubmittedIdDoesNotMatchBase error should have field and type fully expanded
This is just a part of resolving #509.

### PR DESCRIPTION
**Pull request by AstroChelonian**
fixes #509
fixes #528

### CODE REVIEW

File: src/core/api/api_document.pl [hunk 1]
```diff
@@ -155,7 +155,7 @@
 known_document_error(unrecognized_property(_,_,_)).
 known_document_error(casting_error(_,_)).
 known_document_error(submitted_id_does_not_match_generated_id(_,_)).
-known_document_error(submitted_id_does_not_match_base(_,_)).
+known_document_error(submitted_document_id_does_not_have_exp_prefix(_,_)).
```
Comment (matko) [inline src/core/api/api_document.pl:158]: why not just write out exp as expected?

**Review by matko — APPROVED** [review]
yeah seems about right. you could just write 'exp' as 'expected' though for clarity, it's not like we're on a functor length budget.

File: src/core/api/api_document.pl (hunk 1 above)
Comment (spl) [reply src/core/api/api_document.pl:158]: Sure, that's fine.

**Comment by AstroChelonian** [comment]
The error now is the following: 

```
{
    "@type": "api:InsertDocumentErrorResponse",
    "api:error": {
        "@type": "api:SubmittedDocIdDoesNotHaveExpectedPrefix",
        "api:document": {
            "@id": "something",
            "@type": "String"
        },
        "api:prefix": "terminusdb:///data/String/",
        "api:submitted_id": "terminusdb:///data/something"
    },
    "api:message": "Document id 'terminusdb:///data/something' does not have expected prefix 'terminusdb:///data/String/'",
    "api:status": "api:failure"
}

```

**Pull request merged**